### ISSUE TITLE
ApiClients should be de-activatable

### ISSUE THREAD

**Issue by davetron5000** [stitchfix/stitches#25]
When rotating keys, you can't necessarily delete old api keys, for example if you have foreign key constraints to them from your domain objects.

The default `api_clients` table should have an `active` field, defaulting to `true`, or maybe an `active_til` field?  It would probably be handy to have some rake tasks to help with rotation as well.

**Comment by simeonwillbanks** [comment]
@davetron5000 I started adding `active` to `api_clients`.  Please see my initial [commits](https://github.com/stitchfix/stitches/compare/add-api-clients-active).

By adding the [migration template](https://github.com/stitchfix/stitches/compare/add-api-clients-active#diff-13a81ca66f4adca13dfa1a46e66ec813R66), existing *stitches* apps can re-run the stitches generator then migrate.  The generator will copy the `db/migrate/add_active_to_api_clients.rb` file to the apps `migrate` directory with a timestamp prefix.  

```
$ bundle exec rails g stitches:api
...
      create  db/migrate/20160315205935_add_active_to_api_clients.rb
```

After the migration has been copied, we can run migrations.

```
$ bundle exec rake db:migrate
I, [2016-03-15T14:00:49.508379 #13187]  INFO -- : Migrating to AddActiveToApiClients (20160315205935)
== 20160315205935 AddActiveToApiClients: migrating ============================
-- add_column(:api_clients, :active, :boolean, {:default=>true, :null=>false})
   -> 0.0075s
== 20160315205935 AddActiveToApiClients: migrated (0.0076s) ===================
```

Is this how we should implement migrations?  Thanks!

**Comment by davetron5000** [comment]
a) awesome!
b) will running the generator again work or will it try to make a new migration for creating the api_clients migration (because it will have a different timestamp).  I wonder if make a second generator like "upgrade" to add that migration, but also add `active` to the existing migration template?

**Comment by davetron5000** [comment]
This got implemented in 9876c439fe67b565387f645556458b119211e0a4
